qiime2/sdk/util.py

```diff
@@ -92,3 +92,29 @@ def parse_format(format_str):
     except KeyError:
         raise TypeError("No format: %s" % format_str)
     return format_record.format
+
+
+def artifact_actions(string):
+    """Plugins and actions that have as input the artifact type (string)
+
+    Parameters
+    ----------
+    string : str
+        QIIME2 artifact type
+
+    Returns
+    -------
+    list of tuples: [(q2.plugin, [q2.actions, ...]), ...]
+    """
+    commands = []
+
```

Comment:
add something along the lines of this:
```python
query_type = qiime2.sdk.util.parse_type(string)
```
to convert the string into a type. I'm pretty sure it will no-op if its already a type, but I'm not sure.